sympy/utilities/pytest.py

```diff
@@ -90,6 +91,43 @@ def raises(expectedException, code=None):
             raise TypeError(
                 'raises() expects a callable for the 2nd argument.')
 
+    def assert_raise_message(expectedException, err_msg, code=None):
+        """
+        ``assert_raise_message`` is an enhanced version of ``raises``. It tests
+        that error message raised by ``code`` is also same as the expected error
+        message.
+
+        ``assert_raise_message()`` does nothing if the callable raises the expected
+        ``err_msg``, otherwise it raises an AssertionError.
+
+        Examples
+        ========
+
+        >>> from sympy.utilities.pytest import assert_raise_message
+
+        >>> assert_raise_message(ZeroDivisionError, 'zero', lambda: 1/0)
+        >>> assert_raise_message(ZeroDivisionError, 'one', lambda: 1/0)
+        Traceback (most recent call last):
+        ...
+        AssertionError: DID NOT RAISE
+
+        """
+        if code is None:
+            return RaisesContext(expectedException)
```

Comment:
`err_msg` is not checked, when `code` is `None` and `RaisesContext` is used.

Reply:
what `err_msg` can be possibly right when `code` is `None` ?

Reply:
The one raised when `RaisesContext` is exited.